src/Telegram/Endpoints/AvailableMethods.php

```diff
@@ -161,11 +162,31 @@ public function sendVideoNote($video_note, array $opt = []): ?Message
      */
     public function sendMediaGroup(array $media, array $opt = []): ?array
     {
+        $inputMedia = [];
+        $files = [];
+        foreach ($media as $m) {
+            if ($m instanceof InputMedia) {
+                if (is_resource($m->media)) {
+                    $id = uniqid();
+                    $files[$id] = $m->media;
+                    $m->media = "attach://{$id}";
+                }
+            } else {
+                if (is_resource($m['media'])) {
+                    $id = uniqid();
+                    $files[$id] = $m['media'];
+                    $m['media'] = "attach://{$id}";
+                }
```

Comment:
Maybe you can change this code block in this way:
```php
if ($m instanceof InputMedia) {
    if (is_resource($m->media)) {
        $id = uniqid();
        $files[$id] = $m->media;
        $m->media = "attach://{$id}";
    }
} elseif (is_resource($m['media'])) {
    $id = uniqid();
    $files[$id] = $m['media'];
    $m['media'] = "attach://{$id}";
} else {
    continue; //or you can throw an exception
}
```

Reply:
Actually we should not skip it, because the `media` can be also a file_id, so here we want only provide a easier way to attach multiple files

Reply:
You're right.